Code/FooBarTest.php

```diff
@@ -0,0 +1,90 @@
+<?php
+require __DIR__ . '\FooBar.php';
+
+use FooBar;
+use PHPUnit\Framework\TestCase;
+
+class FooBarTest extends TestCase
+{
+    public function testBar()
+    {
+        $Object =  new FooBar();
+        $this->assertEquals("Bar", $Object->Checker(5));
+    }
+
+    public function testFoo()
+    {
+        $Object =  new FooBar();
+        $this->assertEquals("Foo", $Object->Checker(3));
+    }
+
+    public function testFooBar()
+    {
+        $Object =  new FooBar();
+        $this->assertEquals("Foo, Bar", $Object->Checker(15));
+    }
+
+    public function testIntegerToString()
+    {
+        $Object =  new FooBar();
+        $this->assertEquals(13, $Object->Checker(13));
+    }
+
+    public function testNegativeException()
+    {
+        $Object =  new FooBar();
+        $this->expectException(InvalidArgumentException::class);
+        $Object->Checker(-5);
+    }
+
+    public function testNotIntegerException()
+    {
+        $Object =  new FooBar();
+        $this->expectException(InvalidArgumentException::class);
+        $Object->Checker(4.321);
+    }
+
+    public function testNotNumberException()
+    {
+        $Object =  new FooBar();
+        $this->expectException(InvalidArgumentException::class);
+        $Object->Checker("abc");
+    }
+
+    /**
+     * @dataProvider provider
+     */
+
+    public function testMultipleOccurences($expectedResult, $input)
+    {
+        $Object =  new FooBar();
+        $this->assertEquals($expectedResult, $Object->Checker($input));
+    }
+
+    public function provider()
```

Comment:
Does it really need to be 'public'?

Reply:
In the documentation of PHPUnit, there is written - "A data provider method must be **public**..."